ksql-benchmark/src/main/java/io/confluent/ksql/benchmark/SerdeBenchmark.java

```diff
@@ -0,0 +1,196 @@
+ */
+
+package io.confluent.ksql.benchmark;
+
+import static io.confluent.ksql.datagen.DataGenSchemaUtil.getOptionalSchema;
+
+import io.confluent.avro.random.generator.Generator;
+import io.confluent.connect.avro.AvroData;
+import io.confluent.kafka.schemaregistry.client.MockSchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.ksql.GenericRow;
+import io.confluent.ksql.datagen.RowGenerator;
+import io.confluent.ksql.datagen.SessionManager;
+import io.confluent.ksql.serde.avro.KsqlAvroTopicSerDe;
+import io.confluent.ksql.serde.json.KsqlJsonTopicSerDe;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.Pair;
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+import org.apache.avro.Schema;
+
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.common.serialization.Serializer;
+
+import org.apache.kafka.connect.data.Field;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Threads;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+// CHECKSTYLE_RULES.OFF: HideUtilityClassConstructor ; Reason: "@State annotation can only be
+// applied to classes having the default public constructor."
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.MICROSECONDS)
+@Warmup(iterations = 5, time = 10)
+@Measurement(iterations = 5, time = 10)
+@Threads(4)
+public class SerdeBenchmark {
+  // CHECKSTYLE_RULES.ON: HideUtilityClassConstructor
+
+  private static final Path SCHEMA_DIR = Paths.get("schemas");
+  private static final String SCHEMA_FILE_SUFFIX = ".avro";
+  private static final String TOPIC_NAME = "serde_benchmark";
+
+  @State(Scope.Thread)
+  public static class SchemaAndGenericRowState {
+    org.apache.kafka.connect.data.Schema schema;
+    GenericRow row;
+
+    @Param({"impressions", "metrics"})
+    public String schemaName;
+
+    @Setup(Level.Iteration)
+    public void setUp() throws Exception {
+      final Generator generator = new Generator(getSchemaStream(), new Random());
```

Comment:
I tried this but the generated data still varied from run to run. Investigated briefly but couldn't determine why that was the case. (If anyone has thoughts on why this could be happening, I'm all ears!)

From running the benchmarks a few times without seeding `Random()`, it didn't seem like variance was high enough to be of concern, so I'm inclined to leave this as is for now. If while working on the follow-up PR to tune parameters and publish benchmark results variance turns out to be an issue, I'll certainly revisit this.